conan/tools/cmake/base.py

```diff
@@ -162,6 +163,21 @@ def _get_template_context_data(self):
         return ctxt_toolchain, {}
 
     def write_toolchain_files(self):
+        # Warning
+        msg = ("\n*****************************************************************\n"
+               "******************************************************************\n"
+               "'write_toolchain_files()' has been deprecated and moved.\n"
+               "It will be removed in next Conan release.\n"
+               "Use 'generate()' method instead.\n"
+               "********************************************************************\n"
+               "********************************************************************\n")
+        from conans.client.output import Color, ConanOutput
+        ConanOutput(self._conanfile.output._stream,
+                    color=self._conanfile.output._color).writeln(msg, front=Color.BRIGHT_RED)
+        warnings.warn(msg)
```

Comment:
This can be a deprecation warning

```suggestion
        warnings.warn(msg, category=DeprecationWarning)
```